File: examples/java/kora-java-openapi-generator-http-client/src/main/java/ru/tinkoff/kora/example/openapi/http/client/RootService.java
package ru.tinkoff.kora.example.openapi.http.client;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.common.annotation.Root;
import ru.tinkoff.kora.example.openapi.petV2.api.PetApi;

@Root
@Component
public final class RootService {

    private final ru.tinkoff.kora.example.openapi.petV2.api.PetApi petApiV2;
    private final ru.tinkoff.kora.example.openapi.petV3.api.PetApi petApiV3;

    public RootService(PetApi petApiV2, ru.tinkoff.kora.example.openapi.petV3.api.PetApi petApiV3) {
        this.petApiV2 = petApiV2;
        this.petApiV3 = petApiV3;
    }
}

File: examples/kotlin/kora-kotlin-openapi-generator-http-client/src/main/kotlin/ru/tinkoff/kora/kotlin/example/openapi/http/client/RootService.kt
package ru.tinkoff.kora.kotlin.example.openapi.http.client

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.kotlin.example.openapi.petV2.api.PetApi

@Root
@Component
class RootService(
    private val petApiV2: PetApi,
    private val petApiV3: ru.tinkoff.kora.kotlin.example.openapi.petV3.api.PetApi,
)
